### PR TITLE
refactor: move chrono timepoint to/from protobuf timestamp functions to common

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/4376)

### CODE REVIEW

File: google/cloud/internal/time_utils.h [hunk 1]
```diff
@@ -12,33 +12,27 @@
...
-StatusOr<google::protobuf::Timestamp> ConvertTimePointToProtoTimestamp(
-    sys_time<Duration> time_point) {
-  auto const ts = MakeTimestamp(time_point);
-  if (!ts) return std::move(ts).status();
-  return TimestampToProto(*ts);
-}
+std::chrono::system_clock::time_point AsChronoTimepoint(
+    google::protobuf::Timestamp const& ts);
+
+google::protobuf::Timestamp ChronoTimepointToProtoTimestamp(
```
[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/internal/time_utils.h:31]: I agree with @devjgm ... it doesn't help too much to have the source type in the name.

File: google/cloud/internal/time_utils_test.cc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace internal {
+namespace {
+
+TEST(TimeUtils, ConvertTimepointToProtoTimestamp) {
+  auto const epoch = std::chrono::system_clock::from_time_t(0);
+  auto t = epoch + std::chrono::seconds(123) + std::chrono::nanoseconds(456000);
```
Comment (devbww) [inline google/cloud/internal/time_utils_test.cc:28]: Why not `456789`?

File: google/cloud/internal/time_utils.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include "google/cloud/version.h"
+#include <google/protobuf/timestamp.pb.h>
+#include <chrono>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace internal {
+
+std::chrono::system_clock::time_point ProtoTimestampToChronoTimepoint(
```
Comment (devbww) [inline google/cloud/internal/time_utils.h:27]: Don't we want to be able to get full resolution out of the proto (or at least a resolution the user asks for)?

Comment (devbww) [inline google/cloud/internal/time_utils.h:27]: The proto probably has a larger range than `std::chrono::system_clock::time_point`.  What happens when the latter's range is exceeded?

**Comment by devbww** [comment]
> Overall this looks like a good refactoring to me. There may be some all issues that were just noticing now, and for those you can probably just file Issues to unblock yourself.

Yes ... my questions were just that ... questions.  No need to address anything here.

File: google/cloud/internal/time_utils_test.cc [hunk 4]
```diff
@@ -12,30 +12,39 @@
...
+TEST(TimeUtils, ConvertTimepointToProtoTimestamp) {
   auto const epoch = std::chrono::system_clock::from_time_t(0);
   auto t = epoch + std::chrono::seconds(123) + std::chrono::nanoseconds(456000);
-  auto proto_timestamp = ConvertTimePointToProtoTimestamp(t);
-  EXPECT_STATUS_OK(proto_timestamp);
-  EXPECT_EQ(123, proto_timestamp->seconds());
-  EXPECT_EQ(456000, proto_timestamp->nanos());
+  auto proto_timestamp = ChronoTimepointToProtoTimestamp(t);
+  EXPECT_EQ(123, proto_timestamp.seconds());
+  EXPECT_EQ(456000, proto_timestamp.nanos());
```
[the remark this answers is not in the document]
Comment (devjgm) [reply google/cloud/internal/time_utils_test.cc:31]: Nm. I filed https://github.com/googleapis/google-cloud-cpp/issues/4389

**Pull request merged**